### PR TITLE
Feature/movedbclass

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: src/Glpi/Database/MySql.php [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Database;
+
+use DBmysqlIterator;
+use Pdo;
+use Toolbox;
+
```
Comment (cedric-anne) [inline src/Glpi/Database/MySql.php]:
```suggestion
use PDOStatement;

```

File: src/Glpi/Kernel.php [hunk 2]
```diff
@@ -391,9 +391,9 @@ private function loadPluginsServices(ContainerBuilder $container)
      * Else, instanciate a non configure DB to be able to compile container.
      * Returns global variable reference that will be updated if a new connection is established.
      *
-     * @return AbstractDatabase
+     * @return Glpi\Database\AbstractDatabase
      */
-    private function getDbInstance(): AbstractDatabase
+    private function getDbInstance(): Database\AbstractDatabase
```
Comment (cedric-anne) [inline src/Glpi/Kernel.php]: Maybe adding a use statement would be a better solution.

File: src/Glpi/Database/MySql.php (hunk 1 above)
Comment (trasher) [reply src/Glpi/Database/MySql.php]: PDOStatement is not used but in the comments

File: src/Glpi/Kernel.php (hunk 2 above)
Comment (trasher) [reply src/Glpi/Kernel.php]: Done

File: src/Glpi/Database/MySql.php [hunk 3]
```diff
@@ -207,33 +219,6 @@ public function getInfo(): array
...
-    * Is MySQL strict mode ?
-    *
-    * @var DB $DB
-    *
-    * @param string $msg Mode
-    *
-    * @return boolean
-    *
-    * @since 0.90
-    * @deprecated 10.0.0
```
Comment (trasher) [inline src/Glpi/Database/MySql.php]: Is now deprcated since 9.5.0

**Pull request merged**